### PR TITLE
R4R:  Move the types/gov to modules/gov and delete the parameterConfigFile

### PR DESCRIPTION
**Pull request by MrXJC**

### CODE REVIEW

File: modules/gov/codec.go [hunk 1]
```diff
@@ -12,6 +13,15 @@ func RegisterCodec(cdc *codec.Codec) {
 	cdc.RegisterConcrete(MsgSubmitSoftwareUpgradeProposal{}, "gov/MsgSubmitSoftwareUpgradeProposal", nil)
 	cdc.RegisterConcrete(MsgDeposit{}, "cosmos-sdk/MsgDeposit", nil)
```
[the remark this answers is not in the document]
Comment (MrXJC) [reply modules/gov/codec.go]: I see.

File: modules/gov/codec.go [hunk 2]
```diff
@@ -12,6 +13,15 @@ func RegisterCodec(cdc *codec.Codec) {
...
 	cdc.RegisterConcrete(MsgVote{}, "cosmos-sdk/MsgVote", nil)
+	cdc.RegisterInterface((*Proposal)(nil), nil)
+	cdc.RegisterConcrete(&TextProposal{}, "gov/TextProposal", nil)
+	cdc.RegisterConcrete(&govparams.DepositProcedure{}, "cosmos-sdk/DepositProcedure", nil)
+	cdc.RegisterConcrete(&govparams.TallyingProcedure{}, "cosmos-sdk/TallyingProcedure", nil)
+	cdc.RegisterConcrete(&govparams.VotingProcedure{}, "cosmos-sdk/VotingProcedure", nil)
```
Comment (wukongcheng) [inline modules/gov/codec.go]: cosmos-sdk ？

File: modules/gov/codec.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MrXJC) [reply modules/gov/codec.go]: distribution slashing stake are same as gov ?

[the remark this answers is not in the document]
Comment (haifengxi) [reply modules/gov/codec.go]: distribution/slashing/stake can use **stake** as the module name.

**Pull request merged**